src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -72,7 +82,21 @@ public TranslatedNonMatchingResponseBody translateSuccessResponse(List<Assertion
 
         validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
 
+        validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
+
+        AuthnStatement authnStatement = authnAssertion.getAuthnStatements().get(0);
         String nameID = mdsAssertion.getSubject().getNameID().getValue();
+        String issuerId =  authnAssertion.getIssuer().getValue();
+
+        String hashId = userIdHashFactory.hashId(
+                issuerId,
+                nameID,
+                ofNullable(authnStatement.getAuthnContext())
+                        .map(AuthnContext::getAuthnContextClassRef)
+                        .map(AuthnContextClassRef::getAuthnContextClassRef)
+                        .map(authnContextUriValue->getAuthnContextValueFromUri(authnContextUriValue))
```

Comment:
This could be a method reference rather than a lambda, like on the previous lines.  It's probably worth changing that for consistency.

(IntelliJ pointed this out to me!)